src/bridge/bridge.c

```diff
@@ -181,6 +181,8 @@ start_helper_process (const gchar * const  *argv,
       return NULL;
   }
 
+  g_assert (argv[0] != NULL);
```

Comment:
This reads better as a `g_return_val_if_fail()` at the top.

Reply:
That's a trade-off then. AFAIUI `g_return_val_if_fail()` would still apply even with building with `NDEBUG` (which would filter out the asserts). However, it specifically does *not* crash the program, but continues with just a critical warning. I actually *do* want to crash the program, so that our tests have a very specific failure in the sense of "you called start_helper_process() with a bogus argument". Our tests also fail on criticals, but you'd have to look a lot harder to find the root cause.

Reply:
GLib standard approach to this says to use g_return_val_if_fail() to check preconditions on function calls, and turn on abort-on-critical from your tests.

Reply:
Updated.